### DISCUSSION TITLE
State of maintenance for this app

### DISCUSSION THREAD

**Discussion by marcelklehr** [nextcloud/recognize#779] [Announcements]
Hi everyone,

I've posted the following message on a few issues and this raised some questions.

> We are currently maintaining this app on a limited effort basis. This means Nextcloud GmbH will not invest further development resources ourselves in advancing this app with new features. That doesn't mean there will be no new features, however: We do review and enthusiastically welcome community Pull Requests. We would be more than excited if you would like to collaborate with us on this issue. Feel free to reach out here in the comments if you would like to work on this and have questions about how to go about it, how to proceed or would like a short introduction call into the code base. I'm here to help with your questions :v:

So, I'm opening this thread to discuss this and answer questions should you have any.

---

Let me give you a bit of background for this app. I started recognize as a hobby project two years ago, when I was still a student. Last year, after graduating from University, I was hired by Nextcloud GmbH to work on machine learning. My first project was getting recognize production ready and with a lot of help by others in the team we got it to where it is now. It's my pet project and I'm rather proud of what we achieved, so far.

> Can you provide more information regarding Nextcloud GmbH's decision to not continue supporting Recognize? Are they moving away from facial recognition as a feature or using something else? What does this mean for the future of Recognize?

Nextcloud GmbH is not moving away from facial (or any other type of) recognition as a feature and we're not planning to replace recognize with something else. We are also not "dropping support for recognize". It was decided to not spend *employee time* on developing further *features* for this app, *for the time being*.

Nextcloud GmbH has a limited amount of developer time and we need to make strategic decisions about how to use them wisely. Sometimes that means, we have to pause our own plans for feature development for an app, when we've done only 80% of what we'd have liked to have done. **This doesn't mean that we are abandoning this app** (if we did, I'd make that clear here). In software development, there is a distinction between bugs and features. If something doesn't work as intended, it's a bug. If something could be added to make things better, it's a feature. I will continue to fix bugs. But I'm not actively working on implementing new features at the moment. Recognize will be maintained, critical bugs will be fixed and updates released as necessary. We are not abandoning this app and it is still safe to use and rely on it. Meanwhile, we welcome community contributions: We are happy to add new features and steward their development, they just won't be initiated or implemented by us *for the time being*. *For the time being* means that this decision can and hopefully will be subject to change. As the market changes, and Nextcloud GmbH's business opportunities and perspectives change, we may take up development of further features for this app again.

I hope this clears up some things. Feel free to ask your questions here :) 

Cheers
Marcel

**Comment by Andreaux** [comment]
As a UX expert and PO at a Software Development Company I can offer new ideas on how to make the App easier to use and more useful for people, but sadly it is working hands that you need more :) However, I'd be happy to help with suggestions if anyone is willing to contribute coding time. Hit me up if you'd like to work on this together, I really love what this App does and find there's so much untapped potential in there.

**Reply by marcelklehr** [reply]
Thank you for your message @Andreaux If you like, you could hop on our chat room over on gitter to chat about ideas and future development! https://gitter.im/marcelklehr/recognize

I do expect to invest some of my free time on this project as a hobby, so there is the potential to get new things out there :)

**Comment by de-ll** [comment]
I understand that there are strategical decisions to be made. Any PM has to do it. 
However, IMO, there are product features which are like key features which might become even more apparent in the future. I see the recognize app at such a "unicorn" position, e.g. directly competing with Synologys Photos AI features, which draws positive attention from users.
I hope in the future, this decision will be revisited and feature development will continue. 
Thanks for the great app.

**Reply by Andreaux** [reply]
Actually I see tremendous progress with the Recognize app and since I wrote my previous message (a bit further up), the way it is working now is night and day compared to where it was. I am super hyped by all the work @marcelklehr and other people have contributed to Recognize. Have a go at it and see for yourself!
I am using Recognize with the Memories App (the Photos App is a bit behind with features) and it works like a charm. They added a way to see unclassified faces and manually push them around to existing people, etc. I love it.

I fully agree that Recognize could be a major "selling point" for Nextcloud. It surely was for me. I too hope they'll realize this at Nextcloud GmbH.

**Reply by marcelklehr** [reply]
The rub lies in the fact that we need customers that are ready to pay for this app, so we can finance feature development. Our current paying customers often care less about photos solutions in Nextcloud :/

**Comment by bugsyb** [comment]
Here is an idea - would a kickstarter sort of or any other way of funding/subscribing for a feature would be accepted by Nextcloud GmBH? This way this could be turned into a commercial product which would help developing and paying for developers time?
Maybe (this is just maybe) it could be packaged into a product and with a little bit of an effort, i.e. added Coral support and packaged with some small Pi or something creating a beautiful little yet powerful solution for those who want to keep data for ourselves. Even if I wouldn't need a HW, happy to chip into to help getting it done.

**Reply by marcelklehr** [reply]
Mmh, I think Nextcloud GmbH has no interest in becoming an end user facing company. However, anyone can found a company and become a Nextcloud reseller. If you then get a subscription from Nextcloud GmbH for recognize you can sell Nextcloud with recognize and support the development of further features through the subscription.

**Reply by bugsyb** [reply]
Thanks - hopes were that Nextcloud would be interested in getting subscription fees, etc. hence an idea. Thanks!
